project/src/main/java/com/google/sps/data/ProductSearchLibrary.java

```diff
@@ -257,12 +256,47 @@ public static void removeProductFromProductSet(
             
             
         for (ProductSearchResults.Result product : similarProducts) {
-            productIds.add(product.getProduct().getName());
+            
+            String id = getProductId(product.getProduct().getName());
+            productIds.add(id);
         }
-    }
     return productIds;
   }
 
+  public static String getReferenceImage(
+    String productId, 
+    String referenceImageId)
+    throws IOException {
+
+    String imageUri = "";
+    try (ProductSearchClient client = ProductSearchClient.create()) {
+
+        // Get the full path of the reference image.
+        String formattedName =
+            ImageName.format(projectId, computeRegion, productId, referenceImageId);
+        // Get complete detail of the reference image.
+        ReferenceImage image = client.getReferenceImage(formattedName);
+        imageUri = image.getUri();
+    }
+
+    return imageUri;
+  }
+
+  public static void deleteReferenceImage(
+    String productId, 
+    String referenceImageId)
+    throws IOException {
+    try (ProductSearchClient client = ProductSearchClient.create()) {
+
+        // Get the full path of the reference image.
+        String formattedName =
+            ImageName.format(projectId, computeRegion, productId, referenceImageId);
+        // Delete the reference image.
+        client.deleteReferenceImage(formattedName);
+    }
```

Comment:
I think you should either remove the try statement or create a catch so you can tell if deleting the reference image failed or not